Scripts/prepare_release.py

```diff
@@ -57,13 +57,40 @@ def unwrap(opt: Optional[A]) -> A:
     return opt
 
 class NewsFragment(NamedTuple):
-    pr: str
+    pr: Optional[int]
     contents: str
     path: Optional[Path]
 
+    PR_NUM_FORMAT = re.compile(r"[(]#(?P<num>[0-9]+)[)]$", re.MULTILINE)
+
+    @classmethod
+    def resolve_pr_from_git(cls, pth: Path) -> Optional[int]:
+        """Find the PR that introduced `pth`."""
+        proc = git("log", "--diff-filter=A", "--format=oneline", "--", str(pth))
+        if proc.returncode == 0:
+            if m := cls.PR_NUM_FORMAT.search(proc.stdout.strip()):
+                return int(m.group("num"))
+        return None
+
+    @classmethod
+    def resolve_pr(cls, pth: Path) -> Optional[int]:
+        if re.match("^[0-9]+$", pth.stem):
```

Comment:
Hum for many of my news fragments, I used the issue number rather than the PR number in the release. Is that adequate? I think it wouldn't be caught by the script, but at the same time, it feels like linking to the issue for a fix is more important. For a feature however, what you write makes sense.

Reply:
It's all good.  With your README clarification I think we're all set.

Reply:
I actually expanded the README to clarify further.